### DISCUSSION TITLE
New stable release in 2023?

### DISCUSSION THREAD

**Discussion by swenner** [davisking/dlib#2742] [General]
Hi, the latest stable release is nearly a year old (May 2022). I'm waiting for some fixes in master. Is there a plan to release a new stable version any time soon or should users move over to master?

Best, Simon

**Comment by davisking** [comment]
Yeah soon.  Once the new ffmpeg stuff is more settled.

**Reply by rossfg** [reply]
There seems to be a few issues with dlib and explicilty ffmpeg stuff at the moment at least with
ubuntu 22.04.2 using gcc-11.3 

Have also tried on ubuntu 22.10 and 23.04 with gcc-12.2

seems to be a few misplaced const declarations throughout the code!

here is a typical error associated with building the ffmpeg examples code

In file included from /home/rfg/w/dlib/dlib/../dlib/media.h:13,
                 from /home/rfg/w/dlib/examples/ffmpeg_webcam_face_pose_ex.cpp:35:
/home/rfg/w/dlib/dlib/../dlib/media/ffmpeg_demuxer.h: In member function ‘bool dlib::ffmpeg::demuxer::open(const dlib::ffmpeg::demuxer::args&)’:
/home/rfg/w/dlib/dlib/../dlib/media/ffmpeg_demuxer.h:657:73: error: invalid conversion from ‘const AVInputFormat*’ to ‘AVInputFormat*’ [-fpermissive]
  657 | AVInputFormat* input_format = st.args_.input_format.empty() ? nullptr : av_find_input_format(st.args_.input_format.c_str());
      |                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                             |
      |                                                             const AVInputFormat*

/home/rfg/w/dlib/dlib/../dlib/media/ffmpeg_demuxer.h: In lambda function:
/home/rfg/w/dlib/dlib/../dlib/media/ffmpeg_demuxer.h:695:100: error: invalid conversion from ‘AVCodec**’ to ‘const AVCodec**’ [-fpermissive]
  695 | v_find_best_stream(st.pFormatCtx.get(), media_type, -1, -1, &pCodec, 0);
      |                                                             ^~~~~~~
      |                                                             |
      |                                                             AVCodec**
